aten/src/ATen/native/cuda/Shape.cu

```diff
@@ -504,6 +504,11 @@ Tensor& cat_out_cuda(Tensor& out, TensorList inputs, int64_t dimension) {
                 t.device(), " and ", notSkippedTensor->device());
   }
 
+  TORCH_CHECK(
+      out.device() == notSkippedTensor->device(),
+      "All input tensors and out must be on the same device, but inputs are on ",
```

Comment:
Remember to start the error message with the name of the op, so "torch.cat(): all input..."